mm2src/coins/lp_coins.rs

```diff
@@ -1279,10 +1279,14 @@ pub trait WatcherOps {
 
 /// Helper struct wrapping arguments for [TakerCoinSwapOpsV2::send_taker_funding]
 pub struct SendTakerFundingArgs<'a> {
-    /// Taker will be able to refund the payment after this timestamp
-    pub time_lock: u64,
-    /// The hash of the secret generated by taker, this needs to be revealed for immediate refund
+    /// The timestamp after which the taker can refund the funding transaction if the maker hasn't pre-approved the transaction
```

Comment:
I was relying on these conditions in smart contract
```solidity
        if (takerPayments[id].state == TakerPaymentState.TakerApproved) {
            require(
                block.timestamp >= takerPayments[id].paymentLockTime,
                "Current timestamp didn't exceed payment refund lock time"
            );
        }

        if (takerPayments[id].state == TakerPaymentState.PaymentSent) {
            require(
                block.timestamp >= takerPayments[id].preApproveLockTime,
                "Current timestamp didn't exceed payment pre-approve lock time"
            );
        }
``` 

> The funding tx can be refunded even if the maker pre-approved it, only when it's spent and the spending transaction (the taker payment transaction) is confirmed, then it can't be refunded.

I suppose its about `payment_time_lock` when Taker approved tx?

Reply:
Updated doc com for timelocks here https://github.com/KomodoPlatform/komodo-defi-framework/pull/2129/commits/fcd394d0911e29815a140c50acbaf81c6263a0cc